app/controllers/users_controller.rb

```diff
@@ -14,7 +14,9 @@ def create
     using_recaptcha = !params[:spamaway] && Rails.env == "production"
     recaptcha = verify_recaptcha(model: @user) if using_recaptcha
     @spamaway = Spamaway.new(spamaway_params) unless using_recaptcha
-    if ((@spamaway&.valid?) || recaptcha) && @user.save
+
+    saved = @user.save
+    if saved && ((@spamaway&.valid?) || recaptcha)
```

Comment:
Originally when I did not refactor `@user.save` out into a new variable, the ordering of the two statements mattered because the if statement would shortcircuit if recaptcha failed. This is no longer an issue now, and the ordering of the statement doesn't matter.

Reply:
I can change this back if needed

Reply:
If this change is redundant to this pr then please remove this change and edit the issue body.
Thansk

Reply:
I think you forget to remove these redundant changes